### PR TITLE
[JENKINS-51574]  - POM Input support PoC

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-51574

@reviewbybees @raul-arabaolaza

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Also CC @kuisathaverat

File: custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/impl/Builder.java [hunk 1]
```diff
@@ -76,7 +78,21 @@ public void build() throws IOException, InterruptedException {
         if (pathToBom != null) {
             bom = BOM.load(pathToBom);
             LOGGER.log(Level.INFO, "Overriding settings by BOM file: {0}", pathToBom);
-            config.overrideByBOM(bom, config.buildSettings.getEnvironmentName());
+            config.overrideByPOM(bom, config.buildSettings.getEnvironmentName());
```
Comment (raul-arabaolaza) [inline custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/impl/Builder.java]: This should be `overridebyBOM`

File: custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java [hunk 2]
```diff
@@ -161,7 +164,49 @@ public DependencyInfo findPlugin(@Nonnull String artifactId) {
...
+            }
+        } catch (IOException ex) {
+            LOGGER.log(Level.INFO, "Skipping dependency, not an HPI file: " + res);
+            return;
+        }
+
+        plugins.add(res);
+    }
+
+    public void overrideByPOM(@Nonnull BOM bom, @CheckForNull String environmentName) throws IOException {
```
Comment (raul-arabaolaza) [inline custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java]: `overrideByBOM` I guess?

**Comment by raul-arabaolaza** [comment]
I am going to start some work based on this code, mostly related to being sure all dependencies are present in the war when you use a pom file

**Comment by raul-arabaolaza** [comment]
Superseded by #35

**Pull request merged**